src/pymor/algorithms/newton.py

```diff
@@ -102,60 +110,103 @@ def newton(operator, rhs, initial_guess=None, mu=None, error_product=None, least
     U = initial_guess.copy()
     residual = rhs - operator.apply(U, mu=mu)
 
-    err = residual.norm(error_product)[0]
-    logger.info(f'      Initial Residual: {err:5e}')
+    # compute norms
+    solution_norm = U.norm(error_product)[0]
+    solution_norms = [solution_norm]
+    update_norms = []
+    residual_norm = residual.norm(error_product)[0]
+    residual_norms = [residual_norm]
+
+    # select error measure for convergence criteria
+    err = residual_norm if error_measure == 'residual' else solution_norm
+    err_scale_factor = err
+    errs = residual_norms if error_measure == 'residual' else update_norms
+
+    logger.info(f'     res:{residual_norm:.3e}                                 norm:{solution_norm:.3e}')
```

Comment:
Is it clear what "norm" means in this context? I think, if I would see this as an output, I might not be sure what "res" and "norm" mean. "res" would be clear if no "norm" was around.

Reply:
The line is already quite long so I tried to keep the labels as short as possible. Do you have a suggestion how to change the labels?

Reply:
For me, the question is whether or not logging the norm of the (approximate) solutions is actually necessary?

Reply:
It's probably the least important information. On the other hand, I would say that it is often very helpful to know the norm of the solution to get a feeling for what is going on. Obviously, you get the norm also as `upd / rel_upd`, but that is not very convenient.

Maybe the output would be a little less confusing when changing the order to `norm, upd, rel_upd, res, red, tot_red`?

Reply:
> On the other hand, I would say that it is often very helpful to know the norm of the solution to get a feeling for what is going on.

Yes, I agree.

> Maybe the output would be a little less confusing when changing the order to `norm, upd, rel_upd, res, red, tot_red`?

Might be, I'm not sure. As you said, it seems to be the least important information, so putting it in the first place of the output seems a bit contradictory from my point of view.

In general, it's fine to keep `norm`. It really might be helpful sometimes. I could also live with the order you proposed, since I know what is meant by `norm`. However, I'm thinking about users that did not follow our discussion here. I don't know whether having `norm` in the first place of the output might be confusing to them. For instance, if the value of `norm` stagnates somewhere away from 0 (which of course is the case when 0 is not a solution) and the algorithm suddenly stops, saying it's finished due to convergence, it might take some time for the user to realize that `res` fell below the prescribed threshold and thus the algorithm really converged.

Reply:
I changed the order. So it at least is clear that we do not mean the norm of the residual as the residual is mentioned later.

> For instance, if the value of norm stagnates somewhere away from 0 (which of course is the case when 0 is not a solution) and the algorithm suddenly stops, saying it's finished due to convergence, it might take some time for the user to realize that res fell below the prescribed threshold and thus the algorithm really converged.

Note that the algorithm always mentions the reason for convergence.

Let's revisit this after some user feedback.

Reply:
Sounds good!